src/App/app.test.js

```diff
@@ -1,5 +1,23 @@
-import { sum } from './sum';
+import React from 'react';
+import App from './';
+import { mount, shallow } from 'enzyme';
+import { Button } from '@patternfly/react-core';
 
-test('adds 1 + 2 to equal 3', () => {
-  expect(sum(1, 2)).toBe(3);
+describe('App tests', () => {
+  test('should render default App component', () => {
+    const view = shallow(<App />);
+    expect(view).toMatchSnapshot();
+  });
+
+  it('should render a dismiss button', () => {
+    const wrapper = mount(<App />);
+    expect(wrapper.find(Button)).toBeDefined();
+  });
+
+  it('should hide the notification when clicking the dismiss button', () => {
+    const wrapper = mount(<App />);
+    expect(wrapper.state('isShowing')).toEqual(true);
```

Comment:
It is generally bad practice to check state directly. It is better to check that you render correctly based on state. In this case you would asert that the alert is not present.

Also as a heads up. .toBe is more strict and should be used instead of equal if possible. equal is good for comparing objects.